src/Storages/MergeTree/ReplicatedMergeTreeRestartingThread.cpp

```diff
@@ -128,7 +128,10 @@ bool ReplicatedMergeTreeRestartingThread::runImpl()
 
     try
     {
+        Stopwatch watch;
+        LOG_DEBUG(log, "Trying to establish a new connection");
         storage.setZooKeeper();
+        LOG_DEBUG(log, "Establishing a new connection took {} ms", watch.elapsedMilliseconds());
```

Comment:
Better to move it in Context, as the ZooKeeper session is shared between tables.

Reply:
Done